snap/validate.go

```diff
@@ -343,11 +351,28 @@ func Validate(info *Info) error {
 		return err
 	}
 
+	// validate component names, which follow the same rules as snap names
+	for cname, comp := range info.Components {
+		// component Type is validated when unmarshaling
```

Comment:
we need also some kind of ValidateComponentType, which should use the same patter as validPlugSlotIface, until we can switch to a fixed list in some cases

Reply:
But this is actually validated by `ComponentType.UnmarshalYAML`, which is run by `InfoFromSnapYaml` before calls to Validate.  The check currently only considers valid components of type "test" (which can be changed to a regex for the moment if you want).

I know it is a bit strange that validation happens at different types for different types, but the Unmarshaler interface does not let you check everything easily and you get into infinite recursion. I would prefer to actually do the validation always on unmarshalling though, as that happens first. In any case, I've been doing now some tests and something like this could work:

```go
type componentInternal Component

func (ct *Component) UnmarshalYAML(unmarshall func(interface{}) error) error {
	var ci *componentInternal
	// so we avoid recursive calls
	ci = (*componentInternal)(ct)
	if err := unmarshall(ci); err != nil {
		return err
	}

	// now do checks for the different fields
	...
	return nil
}
```
We could check this way Summary and Description, wdyt about moving forward with this approach?

Other option would be to create specific types for Summary and Description, but that seemed a bit too much.

Reply:
As discussed, not doing this ad this is not done by snap.yaml.